docs/modules/ROOT/pages/components.adoc

```diff
@@ -344,6 +344,91 @@ mod MyContract {
 WARNING: Customizing implementations and accessing component storage can potentially corrupt the state, bypass security checks, and undermine the component logic.
 *Exercise extreme caution*. See {security}.
 
+=== Hooks
+
+:hooks-post: https://fleming-andrew.medium.com/extending-cairo-contracts-with-hooks-c3ca21d1d6b8[Extending Cairo Contracts with Hooks]
+
+Hooks are entrypoints to the business logic of a token component that are accessible at the contract level.
+This allows contracts to insert additional behaviors before and/or after token transfers (including mints and burns).
+Prior to hooks, extending functionality required contracts to create <<custom_implementations, custom implementations>>.
+
+All token components include a generic hooks trait.
+When creating a token contract, the using contract must create an implementation of the hooks trait.
+Suppose an ERC20 contract wanted to include Pausable functionality on token transfers.
+The following snippet leverages the `before_update` hook to include this behavior.
+
+[,cairo]
+----
+#[starknet::contract]
+mod MyToken {
+    use openzeppelin::security::pausable::PausableComponent::InternalTrait;
+    use openzeppelin::security::pausable::PausableComponent;
+    use openzeppelin::token::erc20::ERC20Component;
+    use starknet::ContractAddress;
+
+    component!(path: ERC20Component, storage: erc20, event: ERC20Event);
+    component!(path: PausableComponent, storage: pausable, event: PausableEvent);
+
+    // ERC20 Mixin
+    #[abi(embed_v0)]
+    impl ERC20MixinImpl = ERC20Component::ERC20MixinImpl<ContractState>;
+    impl ERC20InternalImpl = ERC20Component::InternalImpl<ContractState>;
+
+    #[abi(embed_v0)]
+    impl PausableImpl = PausableComponent::PausableImpl<ContractState>;
+    impl PausableInternalImpl = PausableComponent::InternalImpl<ContractState>;
+
+    // Create the hooks implementation
+    impl ERC20HooksImpl of ERC20Component::ERC20HooksTrait<ContractState> {
+        // Occurs before token transfers
+        fn before_update(
+            ref self: ERC20Component::ComponentState<ContractState>,
+            from: ContractAddress,
+            recipient: ContractAddress,
+            amount: u256
+        ) {
+            // Access local state from component state
+            let contract_state = ERC20Component::HasComponent::get_contract(@self);
+            // Call function from integrated component
+            contract_state.pausable.assert_not_paused();
+        }
+
+        // Occurs after token transfers
+        fn after_update(
```

Comment:
We now can have default function implementations in traits, so we can simplify this pattern by setting default empty functions in the Hooks trait, removing the need for an empty function in the custom impl. I think we can use this PR for that as well.

Reply:
Note that the default empty impl is still helpful, but it should be refactored to an empty (without functions) impl definition, since the empty functions should come from the Hook trait definition itself.

Reply:
I think this example shouldn't include the empty function, but maybe a note stating that the default for the other hook functions is an empty implementation.